File: Order App/Bidbatl/app/src/main/java/com/android/bidbatl/Activity/SplashActivity.java
package com.android.bidbatl.Activity;

import android.content.Intent;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.os.Handler;

import com.android.bidbatl.Activity.product.MainActivity;
import com.android.bidbatl.R;
import com.preference.PowerPreference;

public class SplashActivity extends AppCompatActivity {
    private final int SPLASH_DISPLAY_LENGTH = 1000;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_splash);
        PowerPreference.getDefaultFile().set("motherPackId","");

        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                /* Create an Intent that will start the Menu-Activity. */
                if (PowerPreference.getDefaultFile().getString("user_id").isEmpty()) {
                    Intent intent = new Intent(getApplicationContext(), FirstActivity.class);
                    startActivity(intent);
                } else {
                    Intent intent = new Intent(getApplicationContext(), MainActivity.class);
                    startActivity(intent);
                }
            }
        }, SPLASH_DISPLAY_LENGTH);
    }
}

File: DeliveryApp/Bidbatl_Delivery/app/src/main/java/com/bidbatl/dileveryapp/SplashActivity.kt
package com.bidbatl.dileveryapp

import android.content.Intent
import android.os.Bundle
import android.os.Handler
import com.bidbatl.dileveryapp.ui.live.HomeActivity
import com.bidbatl.dileveryapp.ui.login.MainActivity
import com.bidbatl.dileveryapp.util.Common

class SplashActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_splash)
        //4second splash time
        Handler().postDelayed({
            if (preference.getBoolean(Common.PreferenceKey.isLoggedIn)){
                navigateToHome()
            }
            else{
                navigateToMain()
            }            //finish this activity
            finish()
        },3000)

    }
    fun navigateToHome() {
        val intent = Intent(BaseApplication.instance, HomeActivity::class.java);
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
        BaseApplication.instance.startActivity(intent)

    }
    fun navigateToMain() {
        val intent = Intent(BaseApplication.instance, MainActivity::class.java);
        intent.flags = Intent.FLAG_ACTIVITY_NEW_TASK
        BaseApplication.instance.startActivity(intent)

    }
}
